_posts/2019-06-27-list-strategy.md

```diff
@@ -0,0 +1,126 @@
+- I want persistent workers for actions that support it, but otherwise only sandboxed execution: `--spawn_strategy=worker,sandboxed`.
+- I want to ensure everything runs remotely: `--spawn_strategy=remote`. Be careful, this configuration will cause a build failure in case bazel finds any action that cannot be executed remotely.
+
+### How to migrate
+You would know that you need to migrate if you see the following error:
+ 
+```
+ERROR: No usable spawn strategy found for spawn with mnemonic %Mnemonic%. Are your --spawn_strategy or --strategy flags too strict?
+```
+
+If that is the case or you want to migrate proactively, see below some ideas on how to fix it.
+
+- Do you use the strategy selection flags `--strategy` or `--spawn_strategy` or `--strategy_regexp`?  
+You'll want to revisit the values you set them to. If you get an error like above, consider adding a fallback strategy to your settings. For example, `--spawn_strategy=sandboxed,local` instead of `--spawn_strategy=sandboxed`.
+
+- Do you currently patch your Bazel version to remove fallback to non-sandboxed execution?  
+This should no longer be necessary then! 
+For example, something like `--spawn_strategy=linux-sandbox` will never fallback to `local` execution anymore.
+
+- Do you execute your build or part of it remotely using `--strategy=remote` and/or `--spawn_strategy=remote`?  
+Now you do not need to specify those strategies anymore - consider removing those `strategy` flags completely! In this case bazel will pickup the first available strategy from the default list, which is `remote`, unless you want to specifically forbid any non-remote executions or to configure custom fallback plan (see next advice).
+
+- If you were using custom fallback strategy for the remote execution, e.g. `--remote_local_fallback_strategy=worker`, just add a fallback strategy to the list, for example: `--spawn_strategy=remote,worker`. 
+Note, `--remote_local_fallback_strategy` flag defaults to `local`, so even if you have never used the flag it was implicitly there for every execution with a `--strategy=remote` flag. If you want to keep the same behavior, now you'd either need to use a default strategy or add a `local` to your custom strategy list: `--spawn_strategy=remote,local`.
+
+- If you are using `--config=remote` in your build you might need to change your .bazelrc file, where all the `remote` configs are located, including `strategy` flags. The above pieces of advice apply to the .bazelrc configurations as well.
+In a simple case, you might completely remove the following lines from .bazelrc (.bazelrc version of [the first example](#example-1):
+
+```
+  build:remote --spawn_strategy=remote
+  build:remote --strategy=Javac=remote
+  build:remote --strategy=Closure=remote
+  build:remote --strategy=Genrule=remote
+  build:remote --remote_local_fallback_strategy=local
+```
+
+or adapt them to your needs:
+
+```
+  build:remote --spawn_strategy=remote,local
```

Comment:
I don't understand the detail and length of this paragraph. I believe the migration story can be summarized as: "Remove any --spawn_strategy and --strategy flags that you might be setting manually or via a .bazelrc file"?

Reply:
simplified